### PR TITLE
Remove legacy code of auto assignment of node ports

### PR DESCRIPTION
**Pull request by anjan-keysight**
@marcushines FYI:
Removing NodePort auto assignment and let Kubernetes handle it, if it is not specified in config file.

### CODE REVIEW

File: topo/node/gobgp/gobgp.go [hunk 1]
```diff
@@ -27,8 +27,6 @@ func New(nodeImpl *node.Impl) (node.Node, error) {
 	if nodeImpl.Proto == nil {
 		return nil, fmt.Errorf("nodeImpl.Proto cannot be nil")
 	}
-	cfg := defaults(nodeImpl.Proto)
```
Comment (shubh90) [inline topo/node/gobgp/gobgp.go:30]: We do not need any defaults in this node ?

Comment (anjan-keysight) [reply topo/node/gobgp/gobgp.go:30]: You are right, we would need those defaults. I will update.

**Comment by anjan-keysight** [comment]
I have added gobgp defaults back.

**Review by shubh90 — APPROVED** [review]

**Review by marcushines — COMMENTED** [review]
can you post results showing this working in a real topology so we can make sure this doesn't cause any regressions?

**Comment by anjan-keysight** [comment]
I have use a 3 ixia node, 2 arista node topology that we normally use for testing. Here's the service data without node_port specified.
keysight@master:~/project/athena/kne$ kubectl get svc -n ixia-c
NAME                   TYPE           CLUSTER-IP       EXTERNAL-IP   PORT(S)                          AGE
gnmi-service           LoadBalancer   10.102.21.101    <pending>     50051:30992/TCP                  23s
grpc-service           LoadBalancer   10.108.152.115   <pending>     40051:30548/TCP                  23s
ixia-c-service         LoadBalancer   10.98.153.23     <pending>     443:31164/TCP                    23s
service-arista1        LoadBalancer   10.110.152.39    <pending>     22:31984/TCP                     23s
service-arista2        LoadBalancer   10.96.4.142      <pending>     22:32724/TCP                     24s
service-ixia-c-port1   LoadBalancer   10.111.239.106   <pending>     50071:32565/TCP,5555:31948/TCP   23s
service-ixia-c-port2   LoadBalancer   10.104.215.160   <pending>     5555:31171/TCP,50071:30946/TCP   24s
service-ixia-c-port3   LoadBalancer   10.96.191.67     <pending>     5555:32320/TCP,50071:31330/TCP   24s

Ran the same test with node_ports 31007 & 31017 assigned to ixia-c-port1 node in config file.
keysight@master:~/project/athena/kne$ kubectl get svc -n ixia-c
NAME                   TYPE           CLUSTER-IP       EXTERNAL-IP   PORT(S)                          AGE
gnmi-service           LoadBalancer   10.104.69.68     <pending>     50051:31790/TCP                  23s
grpc-service           LoadBalancer   10.99.170.10     <pending>     40051:31338/TCP                  23s
ixia-c-service         LoadBalancer   10.97.128.126    <pending>     443:32681/TCP                    23s
service-arista1        LoadBalancer   10.102.248.222   <pending>     22:31234/TCP                     24s
service-arista2        LoadBalancer   10.103.48.235    <pending>     22:32626/TCP                     24s
service-ixia-c-port1   LoadBalancer   10.108.24.59     <pending>     5555:31007/TCP,50071:31017/TCP   23s
service-ixia-c-port2   LoadBalancer   10.100.6.235     <pending>     5555:30506/TCP,50071:31862/TCP   24s
service-ixia-c-port3   LoadBalancer   10.97.184.160    <pending>     5555:32112/TCP,50071:31054/TCP   23s

Also created the same topology twice with only namespace changed. Unique node_ports were assigned.

**Comment by marcushines** [comment]
ty merged

**Pull request merged**